workspaces/util/src/__tests__/chunkBySize-test.ts

```diff
@@ -0,0 +1,78 @@
+import { describe, expect, it } from "@jest/globals";
+import { chunkBySize } from "../chunkBySize";
+
+describe("chunkBySize", () => {
+  it("splits an input array into smaller arrays", () => {
+    const result = chunkBySize([1, 2, 3, 4], 2);
```

Comment:
One small nit I would have is to try to avoid predictable patterns (like 1, 2, 3, 4, ...) in test data. For example, if someone made a big mistake in the implementation, and instead of putting the array elements in the chunks they counted the elements, then that would still pass this test!

```ts
export function* chunkBySize<T>(
  array: readonly T[],
  chunkSize: number,
): Generator<T[], void, void> {
  for (let i = 0; i < array.length; i += chunkSize) {
    const chunk = [];
    for (let j = 1; j <= chunkSize; ++j) {
      // Bug below! Should use array[i + j] instead of i + j,
      // and fix the bounds of j.
      chunk.push(i + j);
    }
    yield chunk;
  }
}
```

Of course this is a bit unlikely, and you also have a test that uses strings which would catch this bug, but it's also an easy enough pattern to avoid as a general principle in test data. The fewer patterns that would result in accidentally passing a test, the better.

If you need inspiration coming up with "random" digits, I'm a big fan of pi: [3, 1, 4, 1, 5, 9, ...] 🥧